src/rosdistro/distribution.py

```diff
@@ -32,22 +32,30 @@
 # POSSIBILITY OF SUCH DAMAGE.
 
 from .manifest_provider.bitbucket import bitbucket_manifest_provider
-from .manifest_provider.git import git_manifest_provider
+from .manifest_provider.git import git_manifest_provider, git_source_manifest_provider
 from .manifest_provider.github import github_manifest_provider
 
 
 class Distribution(object):
 
     default_manifest_providers = [github_manifest_provider, bitbucket_manifest_provider, git_manifest_provider]
+    default_source_manifest_providers = [git_source_manifest_provider]
 
-    def __init__(self, distribution_file, manifest_providers=None):
+    def __init__(self, distribution_file, manifest_providers=None, source_manifest_providers=None):
         self._distribution_file = distribution_file
+
         # Use default
         self._manifest_providers = Distribution.default_manifest_providers
+        self._source_manifest_providers = Distribution.default_source_manifest_providers
+
         # Override default if given
         if manifest_providers is not None:
             self._manifest_providers = manifest_providers
+        if source_manifest_providers is not None:
+            self._source_manifest_providers = source_manifest_providers
```

Comment:
This doubling-up would be removed if the providers were classes with static methods (or we passed the module instead of the function).